linera-core/src/client/mod.rs

```diff
@@ -1542,40 +1534,78 @@ where
 
     /// Downloads and processes from the specified validator a confirmed block certificate that
     /// uses the given blob. If this succeeds, the blob will be in our storage.
-    async fn update_local_node_with_blob_from(
+    async fn update_local_node_with_blobs_from(
         &self,
-        blob_id: BlobId,
+        blob_ids: Vec<BlobId>,
         remote_node: &RemoteNode<P::Node>,
     ) -> Result<(), ChainClientError> {
-        let certificate = remote_node.download_certificate_for_blob(blob_id).await?;
-        // This will download all ancestors of the certificate and process all of them locally.
-        self.receive_certificate(certificate).await?;
+        try_join_all(blob_ids.into_iter().map(|blob_id| async move {
+            let certificate = remote_node.download_certificate_for_blob(blob_id).await?;
+            // This will download all ancestors of the certificate and process all of them locally.
+            self.receive_certificate(certificate).await
+        }))
+        .await?;
+
         Ok(())
     }
 
     /// Downloads and processes a confirmed block certificate that uses the given blob.
     /// If this succeeds, the blob will be in our storage.
-    async fn receive_certificate_for_blob(&self, blob_id: BlobId) -> Result<(), ChainClientError> {
-        let validators = self.validator_nodes().await?;
-        let mut tasks = FuturesUnordered::new();
-        for remote_node in validators {
-            let remote_node = remote_node.clone();
-            tasks.push(async move {
-                let cert = remote_node
-                    .download_certificate_for_blob(blob_id)
-                    .await
-                    .ok()?;
-                self.receive_certificate(cert).await.ok()
-            });
+    pub async fn receive_certificate_for_blob(
+        &self,
+        blob_id: BlobId,
+    ) -> Result<(), ChainClientError> {
+        self.receive_certificates_for_blobs(vec![blob_id]).await
+    }
+
+    /// Downloads and processes confirmed block certificates that use the given blobs.
+    /// If this succeeds, the blobs will be in our storage.
+    pub async fn receive_certificates_for_blobs(
+        &self,
+        blob_ids: Vec<BlobId>,
+    ) -> Result<(), ChainClientError> {
+        let mut validators = self.validator_nodes().await?;
+        let mut tasks = BTreeMap::new();
+
+        for blob_id in blob_ids {
+            if tasks.contains_key(&blob_id) {
+                continue;
+            }
+
+            tasks.insert(blob_id, FuturesUnordered::new());
+            validators.shuffle(&mut thread_rng());
```

Comment:
In `download_certificates` and `download_blob`, it's `LocalNode` that does the shuffling and iterating, not the `ChainClient`. Just for consistency, let's do the same here?